source/beamline/SGM/energy/SGMUndulatorSupport.h

```diff
@@ -94,10 +94,14 @@ inline static bool validEnergy(UndulatorHarmonic undulatorHarmonic, double energ
   * position.
   */
 inline static double optimizedUndulatorPosition(double energy,
-				  SGMUndulatorSupport::UndulatorHarmonic undulatorHarmonic,
-				  double undulatorOffset)
+                                                SGMUndulatorSupport::UndulatorHarmonic undulatorHarmonic,
+                                                double undulatorOffset)
 {
 	double rawUndulatorPosition = (-1/0.14295709668) * log( (1/36.00511212946)*((1737.41045746644/(energy/int(undulatorHarmonic))) -1)) + undulatorOffset;
+
+	if(energy > 1737.41045746644 && undulatorHarmonic == SGMUndulatorSupport::FirstHarmonic) {
```

Comment:
According to the SGM wiki the 0.14295709668, 36.00511212946 and 1737.41045746644 values are all "parametric fit parameters", so I didn't know what to call them other than parametricFit1, parametricFit2 and parametricFit3 which I didn't think was much more helpful.